#!/usr/bin/env python
PACKAGE = "terrain_characterizer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("paint_with_error", bool_t, 0,"Paint each point cloud according to each point error", True)
gen.add("enable_sending", bool_t, 0,"Send out the resulting cloud", True)
gen.add("enable_crop", bool_t, 0,"Crop the cloud into sections", True)
gen.add("total_width", double_t, 0, "Total width of all three subsections combined", 2.0)
gen.add("max_distance", double_t, 0, "Max Distance", 0.05)
gen.add("minY_L", double_t, 0, "MinX_L", -1.0)
gen.add("maxY_L", double_t, 0, "MaxX_L", -0.08)
gen.add("minY_M", double_t, 0, "MinX_L", -1.0)
gen.add("maxY_M", double_t, 0, "MinX_L", -0.08)
gen.add("minY_R", double_t, 0, "MinX_L", -1.0)
gen.add("maxY_R", double_t, 0, "MinX_L", -0.08)
gen.add("minZ", double_t, 0, "MinX_L", -2.5)
gen.add("maxZ", double_t, 0, "MinX_L", +2.5)

exit(gen.generate(PACKAGE, "algorithmParameters", "algorithmParameters"))
